isic/core/migrations/0001_initial_squashed.py

```diff
@@ -0,0 +1,632 @@
+                    django_extensions.db.fields.CreationDateTimeField(
+                        auto_now_add=True, verbose_name='created'
+                    ),
+                ),
+                (
+                    'modified',
+                    django_extensions.db.fields.ModificationDateTimeField(
+                        auto_now=True, verbose_name='modified'
+                    ),
+                ),
+                (
+                    'creator',
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.PROTECT,
+                        related_name='shares',
+                        to=settings.AUTH_USER_MODEL,
+                    ),
+                ),
+                (
+                    'image',
+                    models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='core.image'),
+                ),
+                (
+                    'recipient',
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL
+                    ),
+                ),
+            ],
+            options={
+                'get_latest_by': 'modified',
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='ImageAlias',
+            fields=[
+                (
+                    'id',
+                    models.BigAutoField(
+                        auto_created=True, primary_key=True, serialize=False, verbose_name='ID'
+                    ),
+                ),
+                (
+                    'created',
+                    django_extensions.db.fields.CreationDateTimeField(
+                        auto_now_add=True, verbose_name='created'
+                    ),
+                ),
+                (
+                    'modified',
+                    django_extensions.db.fields.ModificationDateTimeField(
+                        auto_now=True, verbose_name='modified'
+                    ),
+                ),
+                (
+                    'image',
+                    models.ForeignKey(
+                        on_delete=django.db.models.deletion.PROTECT,
+                        related_name='aliases',
+                        to='core.image',
+                    ),
+                ),
+                (
+                    'isic',
+                    models.OneToOneField(
+                        editable=False,
+                        on_delete=django.db.models.deletion.PROTECT,
+                        to='core.isicid',
+                    ),
+                ),
+            ],
+            options={
+                'verbose_name_plural': 'Image aliases',
+            },
+        ),
+        migrations.AddField(
+            model_name='image',
+            name='isic',
+            field=models.OneToOneField(
+                default=isic.core.models.isic_id.IsicId.safe_create,
```

Comment:
This is actually a bug. When this `AddField` is executed by the migration engine, the `default` is evaluated, which has the side effect of inserting rows into the `IsicId` table.

This didn't manifest before because (apparently), an `AlterField` operation to add a `default` doesn't trigger the evaluation. :shrug: 

This made me realize that it's probably a bad idea to use `IsicId.safe_create` as the `default`. Any time an instance of `Image` is created locally, even if it's never saved, a new `IsicId` row is inserted. Instead, I think ideally we should explicitly create both the `IsicId` and the `Image` in the same transaction, so that if `Image` creation fails, the `IsicId` is removed. However, I'm not sure if its possible (or even makes logical sense) to do this in a way that is robust against race conditions in `IsicId` duplicates.

Reply:
This is fixed in #917.